packages/ai-mcp-server/src/node/mcp-theia-server-impl.ts

```diff
@@ -0,0 +1,339 @@
+        }, {
+            capabilities: {
+                tools: {
+                    listChanged: true
+                },
+                resources: {
+                    listChanged: true
+                },
+                prompts: {
+                    listChanged: true
+                }
+            }
+        });
+
+        await this.registerContributions();
+        await this.setupHttpTransport();
+
+        this.running = true;
+    }
+
+    async stop(): Promise<void> {
+        if (!this.running) {
+            return;
+        }
+
+        try {
+            if (this.serverId) {
+                await this.frontendContributionManager.unregisterFrontendContributions(this.serverId);
+            }
+
+            for (const transport of this.httpTransports.values()) {
+                transport.close();
+            }
+            this.httpTransports.clear();
+
+            if (this.httpServer) {
+                await new Promise<void>((resolve, reject) => {
+                    this.httpServer!.close(err => {
+                        if (err) {
+                            reject(err);
+                        } else {
+                            resolve();
+                        }
+                    });
+                });
+                this.httpServer = undefined;
+                this.httpApp = undefined;
+            }
+
+            if (this.server) {
+                this.server.close();
+                this.server = undefined;
+            }
+
+            this.running = false;
+        } catch (error) {
+            this.logger.error('Error stopping MCP server:', error);
+            throw error;
+        }
+    }
+
+    getServer(): McpServer | undefined {
+        return this.server;
+    }
+
+    isRunning(): boolean {
+        return this.running;
+    }
+
+    getServerId(): string | undefined {
+        return this.serverId;
+    }
+
+    protected async setupHttpTransport(): Promise<void> {
+        if (!this.server) {
+            throw new Error('Server not initialized');
+        }
+
+        const config = this.config;
+        if (!config) {
+            throw new Error('Configuration not available');
+        }
+
+        const port = config.port || MCP_SERVER_CONFIG.DEFAULT_PORT;
+        const hostname = config.hostname || MCP_SERVER_CONFIG.DEFAULT_HOSTNAME;
+
+        this.httpApp = express();
+        this.httpApp.use(express.json());
+
+        this.httpApp.use((req, res, next) => {
+            res.header('Access-Control-Allow-Origin', '*');
+            res.header('Access-Control-Allow-Methods', 'GET, POST, DELETE, OPTIONS');
+            res.header('Access-Control-Allow-Headers', 'Content-Type, mcp-session-id');
+            if (req.method === 'OPTIONS') {
+                res.sendStatus(200);
+                return;
+            }
+            next();
+        });
+
+        this.setupHttpEndpoints(this.httpApp);
+        this.httpServer = http.createServer(this.httpApp);
```

Comment:
Do we really need a full separate `httpApp`/`httpServer` on its own port? It would be much nicer if we could just offer it as part of the Theia backend on the `/mcp` route.